specs/src/specs/shares.md

```diff
@@ -0,0 +1,90 @@
+# Shares
+
+<!-- toc -->
+
+## Overview
+
+All available data in a Celestia block is split into fixed-size data chunks known as "shares". A share is associated with exactly one namespace. The shares in a Celestia block are eventually erasure-coded and committed to in [Namespace Merkle trees](./data_structures.md#namespace-merkle-tree).
+
+A share sequence is a contiguous set of shares that contain semantically relevant data. A share sequence may contain one or more shares. In most cases, a share sequence should be parsed together because the original data may have been split across share boundaries. One share sequence exists per [reserved namespace](./consensus.md#reserved-namespaces) and per [blob](../../../x/blob/README.md).
```

Comment:
The "Sparse share" and "compact share" descriptions can also reside under the "Terms and Definitions"  section.

Reply:
Effort was taken to minimize the usage of "sparse share" and "compact share" (one occurence each) because I think it may be easier to frame all shares as sharing the same structure with the added detail that Celestia transaction shares encode their blob uniquely to enable parsing shares in the middle of a sequence.

See https://github.com/celestiaorg/celestia-app/issues/1881

Which other terms should we add to the "Terms and Definitions" section?

Reply:
Started a Glossary section in a1910b5c1d57116bc8781824d71768428c28f756 but I'm on the fence b/c it seems duplicative with terms defined in this doc. Open to adding more terms if you have suggestions.

Reply:
> Effort was taken to minimize the usage of "sparse share" and "compact share" (one occurence each) because I think it may be easier to frame all shares as sharing the same structure with the added detail that Celestia transaction shares encode their blob uniquely to enable parsing shares in the middle of a sequence.
See https://github.com/celestiaorg/celestia-app/issues/1881

I see, thanks for the clarification.

> Started a Glossary section in https://github.com/celestiaorg/celestia-app/commit/a1910b5c1d57116bc8781824d71768428c28f756 but I'm on the fence b/c it seems duplicative with terms defined in this doc. Open to adding more terms if you have suggestions. 

I initially suggested the "Terms and Definitions" section be added at the beginning of the doc in order to introduce and define the key terms that will be repeated throughout the document and allow readers to familiarize themselves with the terminology from the beginning. My suggestion for the "Terms and Definitions" section was motivated by the fact that the "share sequence" has been described early on in the overview part, so figured that if that term is necessary to be introduced early on then it would be better to have a separate section like "Terms and Definitions" to contain all such terms.
On the other hand, a "Glossary" section, which is placed at the end, can also be useful but serves a distinct purpose.

Reply:
Resurrected a terms section.